.github/workflows/deploy.yml

```diff
@@ -0,0 +1,42 @@
+name: deploy
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  deploy:
+    name: Upload release to Test PyPI
+    runs-on: ubuntu-latest
+    environment: pypi
+    permissions:
+      id-token: write # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+      - uses: actions/checkout@v3
+        # this is necessary for setuptools_scm to work properly with github
+        # actions, see https://github.com/pypa/setuptools_scm/issues/480 and
+        # https://stackoverflow.com/a/68959339
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install build
+      - name: Build package
+        run: |
+          python -m build --outdir dist/ --sdist
+      - name: test setuptools_scm
+        run: |
+          pip install setuptools_scm
+          SETUPTOOLS_SCM_DEBUG=1 python -m setuptools_scm
```

Comment:
great, this is very useful! what happens when you run the setuptools_scm in debug mode? what does this do?

Reply:
Just prints out more info, this will eventually be removed -- the PR is not currently ready for review, I just made it a PR so I could trigger the github action. I'll switch it out of draft mode when it's ready for review.